### PR TITLE
Issue 14149 - Bad std.algorithm error messages

### PR DESCRIPTION
**Pull request by sinkuu**
https://issues.dlang.org/show_bug.cgi?id=14149

Added `import std.algorithm.internal : algoFormat;`.

Still the code doesn't work (fails with forward reference error). Blocked by https://issues.dlang.org/show_bug.cgi?id=13372

### CODE REVIEW

**Comment by quickfur** [comment]
I don't like the fact that a submodule will import `std/algorithm/package.d`, which in turn has public imports that pulls in *all* `std.algorithm` submodules. What I originally planned to do was to move `algoFormat` and other internal implementation details into its own submodule, say `std.algorithm.internal`, that only contains stuff that's actually common across all submodules. Then any internal implementation that needs `algoFormat` should `import std.algorithm.internal : algoFormat;` instead of pulling in everything.

**Comment by sinkuu** [comment]
Moved `algoFormat`, `rndstuff`, and `addressOf` to std.algorithm.internal.

File: posix.mak [hunk 1]
```diff
@@ -231,7 +231,8 @@ EXTRA_MODULES += $(EXTRA_DOCUMENTABLES) $(addprefix			\
 	gammafunction errorfunction) $(addprefix std/internal/, \
 	cstring processinit unicode_tables scopebuffer\
 	unicode_comp unicode_decomp unicode_grapheme unicode_norm) \
-	$(addprefix std/internal/test/, dummyrange)
+	$(addprefix std/internal/test/, dummyrange) \
+	$(addprefix std/algorithm/, internal)
```
Comment (e10s) [inline posix.mak]: Why not add it to `STD_ALGO_MODULES`?

Comment (sinkuu) [inline posix.mak]: To prevent document generation.

Comment (e10s) [inline posix.mak]: Ah, docs are generated from the entire stuff in `STD_ALGO_MODULES`...

**Comment by quickfur** [comment]
LGTM, thanks!!

The `rndstuff` code probably belongs in `std.internal.test`, but that probably belongs in another PR.

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
When titling a Pull Request to fix an issue, the title should start with:
```
Fix Issue 14140 - ...
```
Note the `Fix` prefix. The tooling relies on this.

**Comment by sinkuu** [comment]
Yes, I reopened that issue. Sorry for confusing.

**Pull request merged**